sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -487,9 +488,16 @@ private[sbtplugin] object ScalaJSPluginInternal {
               s"$configName := true`")
         }
 
+        if (input.isEmpty) {
+          throw new MessageOnlyException(
+              s"`jsEnvInput in $configName` is empty but `test in $configName` got called. " +
+              "This is not allowed, since running tests requires the generated Scala.js code. " +
+              s"If you want to call `test in $configName` but not have it do anything, " +
+              s"set `test in $configName` := {}`.")
```

Comment:
It's probably time to write scoped tasks using the slash syntax of newer sbt versions, at least in error messages. sbt users tend to only use the slash syntax now, so they might not understand the `in` variants.
```suggestion
              s"`$configName / jsEnvInput` is empty but `$configName / test` got called. " +
              "This is not allowed, since running tests requires the generated Scala.js code. " +
              s"If you want to call `$configName / test` but not have it do anything, " +
              s"set `$configName / test` := {}`.")
```
We should probably swipe through the file to change all the existing error messages at some point, but at least for new messages I think we should use the new syntax.

Reply:
Also consider inverting the two subsentences of the first sentence:
```scala
              s"`$configName / test` got called but `$configName / jsEnvInput` is empty. " +
```
It seems more comprehensible to me, but YMMV.